federation/source-config-handlers/src/main/java/org/codice/ddf/admin/sources/opensearch/OpenSearchSourceUtils.java

```diff
@@ -54,89 +53,67 @@
-            contentType = response.getEntity().getContentType().getValue();
-            if (status == HTTP_OK && OPENSEARCH_MIME_TYPES.contains(contentType) && queryResponse) {
-                return result.trustedCertAuthority(true).certError(false).available(true);
-            } else {
-                return result.trustedCertAuthority(true)
-                        .certError(false)
-                        .available(false);
-            }
-        } catch (SSLPeerUnverifiedException e) {
-            // This is the hostname != cert name case - if this occurs, the URL's SSL cert configuration
-            // is incorrect, or a serious network security issue has occurred.
-            return result.trustedCertAuthority(false)
-                    .certError(true)
-                    .available(false);
-        } catch (Exception e) {
-            try {
-                closeClientAndResponse(client, response);
-                client = getCloseableHttpClient(true);
-                response = client.execute(request);
-                status = response.getStatusLine()
-                        .getStatusCode();
-                contentType = response.getEntity()
-                        .getContentType()
-                        .getValue();
-                DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-                factory.setNamespaceAware(true);
-                DocumentBuilder builder = factory.newDocumentBuilder();
-                Document responseXml = builder.parse(response
-                        .getEntity()
-                        .getContent());
-                queryResponse = (Boolean) xpath.compile(TOTAL_RESULTS_XPATH).evaluate(responseXml, XPathConstants.BOOLEAN);
-                if (status == HTTP_OK && OPENSEARCH_MIME_TYPES.contains(contentType) && queryResponse) {
-                    return result.trustedCertAuthority(false)
-                            .certError(false)
-                            .available(true);
-                }
-            } catch (Exception e1) {
-                return result.trustedCertAuthority(false)
-                        .certError(false)
-                        .available(false);
+            isOpenSearchResponse = (Boolean) xpath.compile(TOTAL_RESULTS_XPATH).evaluate(capabilitiesXml, XPathConstants.BOOLEAN);
+        } catch (XPathExpressionException e1) {
+            return requestResults.addMessage(createCommonSourceConfigMsg(UNKNOWN_ENDPOINT));
+        }
+
+        OpenSearchSourceConfiguration config = new OpenSearchSourceConfiguration();
+        config.sourceUserName(username)
+                .sourceUserPassword(password)
+                .endpointUrl(url)
+                .factoryPid(OPENSEARCH_FACTORY_PID)
+                .configurationHandlerId(OPENSEARCH_SOURCE_CONFIGURATION_HANDLER_ID);
+
+        return requestResults.addMessage(createCommonSourceConfigMsg(isOpenSearchResponse ?
```

Comment:
I wouldn't add the probe result if it is `UNKNOWN_ENDPOINT`. I don't want the UI to ignore the failure message.